reactor-core/src/test/java/reactor/core/publisher/BaseSubscriberTest.java

```diff
@@ -134,35 +136,38 @@ protected void hookFinally(SignalType type) {
 		assertThat(error.get(), is(instanceOf(IllegalStateException.class)));
 	}
 
-	@Test(expected = OutOfMemoryError.class)
+	@Test
 	public void onSubscribeFatalThrown() {
 		Flux<String> flux = Flux.just("foo");
 		AtomicReference<Throwable> error = new AtomicReference<>();
 		AtomicReference<SignalType> checkFinally = new AtomicReference<>();
 
-		flux.subscribe(new BaseSubscriber<String>() {
-			@Override
-			protected void hookOnSubscribe(Subscription subscription) {
-				throw new OutOfMemoryError("boom");
-			}
+		assertThatExceptionOfType(OutOfMemoryError.class)
+				.isThrownBy(() -> {
```

Comment:
knowledge sharing, no action needed:

when it comes to `assertThatExceptionOfType` + `isThrownBy`, sometimes it makes sense to put `isThrownBy`
at the same line as `assertThatExceptionOfType`. It will result in much less indentation of lambda's body :)

Reply:
That's actually worth doing I think. Let me try that.